eric_sse/entities.py

```
@@ -1,55 +1,64 @@
+        del self.__queues[listener_id]
+        del self.__listeners[listener_id]
+        self.__queues_factory.delete(listener_id)
 
-    def stop_sync(self) -> None:
-        logger.debug(f"Stopping listener {self.id}")
-        self.__is_running = False
+    def get_queue(self, listener_id: str) -> Queue:
+        try:
+            return self.__queues[listener_id]
+        except KeyError:
+            raise InvalidListenerException(f"Invalid listener {listener_id}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)